Source/WebKit/UIProcess/mac/WebPageProxyMac.mm

```diff
@@ -182,12 +182,20 @@ static inline bool expectsLegacyImplicitRubberBandControl()
 
 void WebPageProxy::windowAndViewFramesChanged(const FloatRect& viewFrameInWindowCoordinates, const FloatPoint& accessibilityViewCoordinates)
 {
-    if (!hasRunningProcess())
-        return;
-
     // In case the UI client overrides getWindowFrame(), we call it here to make sure we send the appropriate window frame.
     m_uiClient->windowFrame(*this, [this, protectedThis = Ref { *this }, viewFrameInWindowCoordinates, accessibilityViewCoordinates] (FloatRect windowFrameInScreenCoordinates) {
         FloatRect windowFrameInUnflippedScreenCoordinates = pageClient().convertToUserSpace(windowFrameInScreenCoordinates);
+
+        if (!hasRunningProcess()) {
+            m_viewWindowCoordinates = makeUnique<ViewWindowCoordinates>();
+            auto& coordinates = *m_viewWindowCoordinates;
+            coordinates.windowFrameInScreenCoordinates = windowFrameInScreenCoordinates;
+            coordinates.windowFrameInUnflippedScreenCoordinates = windowFrameInUnflippedScreenCoordinates;
+            coordinates.viewFrameInWindowCoordinates = viewFrameInWindowCoordinates;
+            coordinates.accessibilityViewCoordinates = accessibilityViewCoordinates;
+            return;
+        }
+
         send(Messages::WebPage::WindowAndViewFramesChanged(windowFrameInScreenCoordinates, windowFrameInUnflippedScreenCoordinates, viewFrameInWindowCoordinates, accessibilityViewCoordinates));
```

Comment:
Why not change this message to send the new ViewWindowCoordinates struct?

Reply:
Sure, we can do that.